api/client/client.go

```diff
@@ -4766,3 +4761,48 @@ func (c *Client) UpsertUserPreferences(ctx context.Context, in *userpreferencesp
 func (c *Client) ResourceUsageClient() resourceusagepb.ResourceUsageServiceClient {
 	return resourceusagepb.NewResourceUsageServiceClient(c.conn)
 }
+
+// UpdateRemoteCluster updates remote cluster from the specified value.
+// TODO(noah): In v17.0.0 this method should switch to call UpdateRemoteCluster
+// on the presence service client.
+func (c *Client) UpdateRemoteCluster(ctx context.Context, rc types.RemoteCluster) error {
+	rcV3, ok := rc.(*types.RemoteClusterV3)
+	if !ok {
+		return trace.BadParameter("unsupported remote cluster type %T", rcV3)
+	}
+
+	_, err := c.grpc.UpdateRemoteCluster(ctx, rcV3)
+	return trace.Wrap(err)
+}
+
+// ListRemoteClusters returns a page of remote clusters.
+func (c *Client) ListRemoteClusters(ctx context.Context, pageSize int, nextToken string) ([]types.RemoteCluster, string, error) {
+	res, err := c.PresenceServiceClient().ListRemoteClusters(ctx, &presencepb.ListRemoteClustersRequest{
+		PageSize:  int32(pageSize),
+		PageToken: nextToken,
+	})
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	var rcs []types.RemoteCluster
+	for _, rc := range res.RemoteClusters {
+		rcs = append(rcs, rc)
+	}
```

Comment:
Why don't we return `res.RemoteClusters` directly?

Reply:
So the type of `res.RemoteClusters` is `[]*types.RemoteClusterV3`. `*types.RemoteClusterV3` implements `types.RemoteCluster` but unfortunately in Go, you still can't return `[]*types.RemoteClusterV3` as `[]types.RemoteCluster` - https://stackoverflow.com/questions/12753805/type-converting-slices-of-interfaces/12754757#12754757